Put all resources in specified provider namespace

lblackstone opened the pull request:
This is an updated version of #506 that should correctly handle CRs.

Fixes #415

lblackstone commented:
Error is:
```
kubernetes:config.istio.io:rule istio-system/promhttp  error: kubernetes:config.istio.io/v1alpha2:rule resource 'istio-system/promhttp' has a problem:  Found API Group, but it did not contain a schema for "config.istio.io/v1alpha2, Kind=rule"
```

Not sure yet if that's legit, or if https://github.com/pulumi/pulumi-kubernetes/pull/538/files#diff-6155820fac2dd2fda7353505b1f69319R307 is invalid.

metral commented on pkg/clients/clients.go:
```diff
@@ -132,10 +132,15 @@ func (dcs *DynamicClientSet) gvkForKind(kind kinds.Kind) (*schema.GroupVersionKi
 
-func (dcs *DynamicClientSet) namespaced(gvk schema.GroupVersionKind) (bool, error) {
-	resourceList, err := dcs.DiscoveryClientCached.ServerResourcesForGroupVersion(gvk.GroupVersion().String())
+func (dcs *DynamicClientSet) NamespacedKind(gvk schema.GroupVersionKind) (bool, error) {
```
nit: s/NamespacedKind/IsNamespacedKind

lblackstone commented:
@metral I think I addressed all your feedback. Added a note to the code clarifying the override behavior.

CRs are a special case because it isn't always possible to tell if they are Namespaced/Global scoped during preview. As noted in the code, this could result in a namespace being set erroneously on a Cluster-scoped CR for previews, but this would be corrected during the Create step, when the required information is available. This correction will be reflected in further updates/diffs as well, so even if the erroneous namespace on the resource changes, it will not require a replacement.

lblackstone commented:
@hausdorff Have you had a chance to look this over? The main change from the previously-merged #506 is the improved logic related to unregistered CRs. See the explanation in my comment to Mike for details.
